tests/unit/unit-term-id.cpp

```diff
@@ -0,0 +1,86 @@
+
+/*********************                                           */
+/*! \file unit-term-id.cpp
+** \verbatim
+** Top contributors (to current version):
+**   Makai Mann
+** This file is part of the smt-switch project.
+** Copyright (c) 2020 by the authors listed in the file AUTHORS
+** in the top-level source directory) and their institutional affiliations.
+** All rights reserved.  See the file LICENSE in the top-level source
+** directory for licensing information.\endverbatim
+**
+** \brief Unit tests for Term unique ids
+**
+**
+**/
+
+#include <utility>
+#include <vector>
+
+#include "available_solvers.h"
+#include "gtest/gtest.h"
+#include "smt.h"
+
+using namespace smt;
+using namespace std;
+
+namespace smt_tests {
+
+class UnitTermIdTests
+    : public ::testing::Test,
+      public ::testing::WithParamInterface<SolverConfiguration>
+{
+ protected:
+  void SetUp() override
+  {
+    s = create_solver(GetParam());
+    s->set_opt("produce-models", "true");
+    s->set_opt("incremental", "true");
+
+    boolsort = s->make_sort(BOOL);
+    bvsort = s->make_sort(BV, 4);
+    a = s->make_symbol("a", boolsort);
+    b = s->make_symbol("b", boolsort);
+    x = s->make_symbol("x", bvsort);
+    y = s->make_symbol("y", bvsort);
+    one = s->make_term(1, bvsort);
```

Comment:
can we make another term with the same constant value. 

`second_one = s->make_term(1, bvsort);`

and check its equivalence with `one`.

Does that make sense?

Reply:
Yeah that's a great idea, thanks! Added with dc744e79c17f011b3877225118d8f177af359d05